encrypted-media-respec.html

```diff
@@ -3779,17 +3779,13 @@ <h2>
                       </li>
                       <li>
                         <p>
-                          Run the [=Queue a "message" Event=] algorithm on the <var>session</var>,
-                          providing <var>message type</var> and <var>message</var>.
+                          Resolve <var>promise</var>.
```

Comment:
Like.. this ... (please don't merge this, just an example):

```suggestion
                          , [queue a global task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-a-global-task) on the [??? task source](https://html.spec.whatwg.org/multipage/webappapis.html#user-interaction-task-source) using global to resolve <var>promise</var> with `undefined`.
```

Reply:
Note that promises must resolved with something... even is it's `undefined`

Reply:
Okay, now that I know what to look for, I see lots of examples of "resolve promise with undefined" in other Media WG specs.  Thank you for pointing that out!

I don't see anything like "queue a global task on ... to resolve promise" in any of our other specs.  I see examples of "resolve" in the steps of a queued task, or in the steps of something run "in parallel".  In fact, this bit you're commenting on is already inside of "[=Queue a task=] to run the following steps".  So it seems like this part of your suggestion is maybe already covered.  Please let me know if I misunderstood.

I'll update all the resolutions to include a value, even if it's `undefined`.

Reply:
Yeah, all our specs are wrong 😭 we need to fix them.

Reply:
We can fix things incrementally though. 

They way to understand this is that something in parallel happens off main thread, then with the result of the asynchronous operation we queue a task (on the main thread, using an explicit task queue) to resolve that promise with some value.